x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/optimizer/OptimizerRunTests.java

```diff
@@ -48,4 +57,57 @@ public void testWhereClause() {
         LogicalPlan p = plan("SELECT some.string l FROM test WHERE int IS NOT NULL AND int < 10005 ORDER BY int");
         assertNotNull(p);
     }
-}
\ No newline at end of file
+
+    public void testReducedBinaryComparisonGreaterThenOrEqual() {
```

Comment:
I'd like to see more tests that the behavior regarding commutative vs non-commutative expressions:
```
a + 2 > 3
2 + a > 3
a - 2 > 3
2 - a > 3
a * 2 > 3
2 * a > 3
a / 2 > 3
2 / a > 3
1 / (a - 2) > 3
1 / (2 - a) > 3
1 / (2 + (2 - a) > 3
```

Reply:
Added all these tests.